src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -59,12 +68,85 @@ def _validation(target, data, evaluators, output_path, tracking_uri, evaluation_
+    return initial_data_df
+
+
+def _validate_columns(df: pd.DataFrame, evaluators: Dict[str, Any], target: Optional[Callable]) -> None:
+    """
+    Check that all columns needed by evaluator or target function are present.
+
+    :keyword df: The data frame to be validated.
+    :paramtype df: pd.DataFrame
+    :keyword evaluators: The dictionary of evaluators.
+    :paramtype evaluators: Dict[str, Any]
+    :keyword target: The callable to be applied to data set.
+    :paramtype target: Optional[Callable]
+    """
+    if target:
+        # If the target function is given, it may return
+        # several columns and hence we cannot check the availability of columns
+        # without knowing target function semantics.
+        # Instead, here we will validate the columns, taken by target.
+        _validate_input_data_for_evaluator(target, None, df, is_target_fn=True)
+    else:
+        for evaluator_name, evaluator in evaluators.items():
+            _validate_input_data_for_evaluator(evaluator, evaluator_name, df)
+
+
+def _apply_target_to_data(
+        target: Callable,
+        data: str,
+        pf_client: PFClient,
+        initial_data: pd.DataFrame) -> Tuple[str, pd.DataFrame, List[str]]:
+    """
+    Apply the target function to the data set and save the data to temporary file.
+
+    :keyword target: The function to be applied to data.
+    :paramtype target: Callable
+    :keyword data: The path to input jsonl file.
+    :paramtype data: str
+    :keyword pf_client: The promptflow client to be used.
+    :paramtype pf_client: PFClient
+    :keyword initial_data: The data frame with the loaded data.
+    :paramtype initial_data: pd.DataFrame
+    :return: The tuple, containing path, data frame and the list of added columns.
+    :rtype: Tuple[str, pd.DataFrame, List[str]]
+    """
+    # We are manually creating the temporary directory for the flow
+    # because the way tempdir remove temporary directories will
+    # hang the debugger, because promptflow will keep flow directory.
+    run = pf_client.run(
+        flow=target,
+        data=data,
+        name=f'preprocess_{uuid.uuid1()}',
+        stream=True
+    )
+    target_output = pf_client.runs.get_details(run, all_results=True)
+    # Remove input and output prefix
+    re_prefix = re.compile('(outputs[.])|(inputs[.])')
+    added_columns = {
+        re_prefix.sub('', col) for col in filter(
+            lambda x: x.startswith('outputs.'), target_output.columns)}
+
+    rename_dict = {col: re_prefix.sub('', col) for col in target_output.columns}
+    target_output.rename(columns=rename_dict, inplace=True)
+    drop_columns = set(target_output.columns) - added_columns
+    target_output.set_index(LINE_NUMBER, inplace=True)
+    target_output.sort_index(inplace=True)
+    target_output.reset_index(inplace=True, drop=False)
+    # target_output contains only input columns, taken by function,
+    # so we need to concatenate it to the input data frame.

```

Comment:
Can we just get the columns with outputs. prefix and concat with initial_data_df ?

Reply:
Fixed.